server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -369,7 +369,8 @@ public ClusterState addComponentTemplate(
         }
 
         if (finalComponentTemplate.template().lifecycle() != null) {
-            finalComponentTemplate.template().lifecycle().addWarningHeaderIfDataRetentionNotEffective(globalRetentionSettings.get());
+            // We do not know if this lifecycle will belong to an internal data stream, so we fall back to a non internal.
```

Comment:
Do we have any system data streams that define a retention? I'm just wondering if this is going to cause problems?

Reply:
We don't currently (we only have one, and it doesn't have a retention). I think even if we did though, there wouldn't be anywhere to add the header, since the request is coming from inside of ES itself?

Reply:
What if we checked if the index pattern of the composable template is dot-prefixed? It's not full-proof but it might reduce the noise?

Reply:
@dakrone & @masseyke I gave it a shot. If there are no objections I will merge this.